shopfloor_base/models/shopfloor_scenario.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2021 ACSONE SA/NV (http://www.acsone.eu)
+# @author Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+
+from odoo import api, fields, models
+
+from odoo.addons.base_sparse_field.models.fields import Serialized
+from odoo.addons.http_routing.models.ir_http import slugify
+
+_logger = logging.getLogger(__name__)
+
+try:
+    import yaml
+except ImportError:
+    _logger.debug("`yaml` lib is missing")
+
+
+class ShopfloorScenario(models.Model):
+    _name = "shopfloor.scenario"
+    _description = "Shopfloor Scenario"
+
+    name = fields.Char(required=True, translate=True)
+    key = fields.Char(required=True)
+    options = Serialized(compute="_compute_options", default={})
+    options_edit = fields.Char(help="Configure options via YAML")
```

Comment:
yaml ? 
Why not a sparse field like in storage ?

Reply:
It is a sparse field BUT this gives you much more flexibility in adding tech configurations w/out relying or real fields.
I find it very handy.

Reply:
Introducing a whole lib just for that, it's a bit overkill no ?

It's less hype but JSON does job.

Reply:
JSON sucks for manual editing. Adding pyyaml is not overkilling IMO.
It's an easy way to provide an easy way to edit options w/out getting insane w/ your own parser or hacks to display the values in a nice way (eg: body/body_str in se.index.config).

Reply:
From an IT perspective : less dependencies = less conflicts + easier install

YAML has it's drawbacks too. It's still an hacky way to set things up, it's fine for the moment and I don't block it's addition even if I think it's not the good place to put it.

One more question: what are the extra settings you want to set?

Reply:
Extra settings can be whatever you want. They are translated into `self.options` in services.
You see an example here on how it can be used https://github.com/OCA/wms/blob/13.0/shopfloor/services/checkout.py#L79
Here you see the config for existing scenario https://github.com/simahawk/wms/blob/shopfloor-split/shopfloor/data/shopfloor_scenario_data.xml#L5
The point is that these options should be "dynamic" and ppl should be able to use them as they prefer, even with nested values or list of values, etc.

That been said, I'll remove YAML and rely on good writing by configurators :)